src/Common/Commands.Common.Authentication/Factories/AuthenticationFactory.cs

```diff
@@ -317,41 +298,47 @@ public ServiceClientCredentials GetServiceClientCredentials(IAzureContext contex
                 }
 
                 ServiceClientCredentials result = null;
-
-                if (context.Account.Type == AzureAccount.AccountType.User)
-                {
-                    result = Rest.Azure.Authentication.UserTokenProvider.CreateCredentialsFromCache(
-                        AdalConfiguration.PowerShellClientId,
-                        tenant,
-                        context.Account.Id,
-                        env,
-                        tokenCache as TokenCache).ConfigureAwait(false).GetAwaiter().GetResult();
-                }
-                else if (context.Account.Type == AzureAccount.AccountType.ServicePrincipal)
-                {
-                    if (context.Account.IsPropertySet(AzureAccount.Property.CertificateThumbprint))
-                    {
-                        result = ApplicationTokenProvider.LoginSilentAsync(
-                            tenant,
-                            context.Account.Id,
-                            new CertificateApplicationCredentialProvider(
-                                context.Account.GetThumbprint()),
-                            env,
-                            tokenCache as TokenCache).ConfigureAwait(false).GetAwaiter().GetResult();
-                    }
-                    else
-                    {
-                        result = ApplicationTokenProvider.LoginSilentAsync(
-                            tenant,
-                            context.Account.Id,
-                            new KeyStoreApplicationCredentialProvider(tenant),
-                            env,
-                            tokenCache as TokenCache).ConfigureAwait(false).GetAwaiter().GetResult();
-                    }
-                }
-                else
+                switch (context.Account.Type)
                 {
-                    throw new NotSupportedException(context.Account.Type.ToString());
+                    case AzureAccount.AccountType.ManagedService:
```

Comment:
@markcowl just so I understand, if you previously tried to get the service client credentials for a user who logged in via MSI, then this would throw a `NotSupportedException`, but isn't the call to get the credentials made during [`BeginProcessing`](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Common/Commands.ResourceManager.Common/AzureRMCmdlet.cs#L290) for every ARM cmdlet? How did this work previously for users?

Reply:
It depends on which kind of client they were using, and how they did authentication.  For the lion's share of ARM clients, this would not have worked.